code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstanceStart.java

```diff
@@ -117,6 +129,58 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
         return result;
     }
 
+    protected void waitForDependenciesStart(Instance instance) {
+        List<Long> instancesIds = DataAccessor.fieldLongList(instance, DockerInstanceConstants.FIELD_VOLUMES_FROM);
+        Long networkFromId = DataAccessor.fieldLong(instance, DockerInstanceConstants.FIELD_NETWORK_CONTAINER_ID);
+        if (networkFromId != null) {
+            instancesIds.add(networkFromId);
+        }
+        List<Instance> waitList = new ArrayList<>();
+        for (Long id : instancesIds) {
+            Instance i = objectManager.loadResource(Instance.class, id);
+            List<String> removedStates = Arrays.asList(CommonStatesConstants.REMOVED, CommonStatesConstants.REMOVING);
+            List<String> stoppedStates = Arrays.asList(InstanceConstants.STATE_STOPPED, InstanceConstants.STATE_STOPPING);
+            String type = networkFromId != null && networkFromId.equals(id) ? "networkFrom" : "volumeFrom";
+            if (removedStates.contains(i.getState())) {
+                throw new ExecutionException("Dependencies readiness error ", type + " instance is removed", i);
```

Comment:
@ibuildthecloud throwing executionException as situation when data/network container is not present == docker error when non-existing container id is passed in. Let me know if you think there should be some other exception.